### PR TITLE
Add ApplicationFactory

### PR DESCRIPTION
**Pull request by rfelgent**
Hi @joshiste,

here the PR for #331

### CODE REVIEW

**Comment by joshiste** [comment]
Looking good so far. Could you do a rebase and a forced push to resolve conflicts and to update the PR?

**Comment by rfelgent** [comment]
I try my best to resolve the conflicts ;-)

Pls have a look at my FIXME and TODO hint.

File: spring-boot-admin-starter-client/src/test/java/de/codecentric/boot/admin/services/ApplicationFactoryTest.java [hunk 1]
```diff
@@ -24,7 +25,8 @@
 import org.springframework.context.annotation.Configuration;
 import org.springframework.web.context.support.AnnotationConfigWebApplicationContext;
 
-public class AdminClientPropertiesTest {
+//FIXME: do test on methods with visibility public, but not protected ...
```
Comment (joshiste) [inline spring-boot-admin-starter-client/src/test/java/de/codecentric/boot/admin/services/ApplicationFactoryTest.java]: Yeah the test should be all about the public methods. Should be no problem to test the behaviour of the private methods via the public api....

File: spring-boot-admin-starter-client/src/main/java/de/codecentric/boot/admin/services/ApplicationFactory.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    public ApplicationFactory(AdminClientProperties client, ManagementServerProperties management, ServerProperties server) {
+        this.client = client;
+        this.management = management;
+        this.server = server;
+    }
+
+    public Application createApplication() {
+        //Remember, that not all information are set by the factory, e.g. id and status is set by ApplicationRegistry
+        //TODO: rethink about the co-existence of builder and factory pattern for the domain class Application (Appliation.Builder and ApplicationFactory)!
```
Comment (joshiste) [inline spring-boot-admin-starter-client/src/main/java/de/codecentric/boot/admin/services/ApplicationFactory.java]: I don't have a problem with having a builder and a factory.
The former is just for ease of development/reading code and the latter on having (multiple) strategies on how to construct an object.

File: spring-boot-admin-starter-client/src/test/java/de/codecentric/boot/admin/services/ApplicationFactoryTest.java (hunk 1 above)
Comment (rfelgent) [inline spring-boot-admin-starter-client/src/test/java/de/codecentric/boot/admin/services/ApplicationFactoryTest.java]: I will fix this tomorrow.

File: spring-boot-admin-starter-client/src/main/java/de/codecentric/boot/admin/services/ApplicationFactory.java (hunk 2 above)
Comment (rfelgent) [inline spring-boot-admin-starter-client/src/main/java/de/codecentric/boot/admin/services/ApplicationFactory.java]: ok, i will remove the "TODO:".

But I would suggest to keep my first comment, as I had difficulties to understand, that the created Application object by the ApplicationFactory is later on manipulated (by copy) with id/status by ApplicationRegistry. 
The comment helps me (and maybe others) do not forget this.

**Comment by joshiste** [comment]
merged with 60c027e2a337df61d591bf620bf360188c39e2ba

**Pull request closed without merging**